nodejs/week5/src/frontend/pages/meal.js

```diff
@@ -0,0 +1,117 @@
+      Location : ${data[0].location}  <br>
+      Date & Time : ${data[0].when_date} <br>
+      Max Guests : ${data[0].max_reservations} <br>
+      Cost : dkk ${data[0].price}`;
+      document.getElementById("root").appendChild(newListElement);
+
+      //injecting form for reservation
+      document.getElementById("form").innerHTML = `
+      <button id="cmdShowForm" class="btn-primary btn-lg" disabled> Sorry.. FULLY BOOKED </button>
+      <form id="reservationForm" class="ReservationForm">
+ <h3><strong>Reservation Information</strong></h3>     
+<input type="text"  class="form-control" id="name" name="name" placeholder="Contact Name">
+<br>
+<input type="text" class="form-control" id="phone" name="phone" placeholder = "Conatact Number">
+<br>
+<input type="text" class="form-control" id="guestCount" name="guestCount" placeholder = "Number of Guests">
+<br>
+      
+<button id="cmdReservation" class="btn-danger btn-lg"> Save Reservation </button> 
+
+</form>`;
+
+      fetch("api/availabelreservations")
+        .then(res => res.json())
+        .then(data => {
+          data.forEach(el => {
+            if (el.id == id && el.max_reservations > el.total_reservations) {
+              document.getElementById("cmdShowForm").disabled = false;
+              document.getElementById("cmdShowForm").textContent = "Click for Reservations";
+            }
+
+            //button.visibility="visible";
+          })
+        });
+
+      document.getElementById("cmdShowForm").addEventListener("click", () => {
+        document.getElementById("reservationForm").style.visibility = "visible";
+      })
+      document.getElementById("cmdReservation").addEventListener("click", (event) => {
+        event.preventDefault();
+        const meal_id = id;
+        const name = document.getElementById("name").value;
+        const phone = document.getElementById("phone").value;
+        const number_of_guests = document.getElementById("guestCount").value;
+         //const created_date = new Date();
+        const newReservation = {
+          number_of_guests,
+          meal_id,
+          //created_date,
+          phone,
+          name
+
+        }
+
+
+        // // function checking only alphabets in string
+        // function checkstring(str) {
+        //   var alphbets = new RegExp(/^[a-zA-Z]{3,}$/);
+        //   if (str.test(alphbets)) {
+        //     return true;
+        //   }
+        //   else {
+        //     alert("message");
+        //     return false;
+        //   }
+        // }
+        // ////////////////////////////////////////
+
+        // if (!checkstring(name)) {
+        //   alert("Name must be String value .....");
+        //   document.getElementById("name").value = "";
+        //   return;
+        // }
+
+        if (!Number.isInteger(parseInt(number_of_guests))) {
+          alert("No of reservations must be Integer....");
+          document.getElementById("guestCount").value = "";

```

Comment:
Avoid using `getElementById` instead use `querySelector`. `getElementById` forces you to use id's which is a bad practice